lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -1889,26 +1889,50 @@ def cmd_db_disconnect(*args)
     end
   end
 
-  def cmd_db_rebuild_cache
-    unless framework.db.active
-      print_error("The database is not connected")
-      return
-    end
-
-    print_status("Purging and rebuilding the module cache in the background...")
-    framework.threads.spawn("ModuleCacheRebuild", true) do
-      framework.db.purge_all_module_details
-      framework.db.update_all_module_details
-    end
-  end
-
   def cmd_db_rebuild_cache_help
     print_line "Usage: db_rebuild_cache"
     print_line
-    print_line "Purge and rebuild the SQL module cache."
+    print_line "Rebuild the local module data store"
+    print_line
+    print_line "   OPTIONS:"
+    print_line "       -h, --help          Help"
+    print_line "       -u, --update-base   Update the base data store (developers only)"
     print_line
   end
 
+  def cmd_db_rebuild_cache(*args)
+    update_base = false
+    while (arg = args.shift)
```

Comment:
I would prefer this to use the `OptionParser` as in https://github.com/rapid7/metasploit-framework/pull/10345 to have a more extensible and less fragile option parsing.

Reply:
Please just do it, don't wait for me!

Reply:
I was just using the style every other command in this file uses (not really interested in updating everything there, but surely someone could).